firmware/signing.c

```diff
@@ -990,12 +1016,40 @@ void signing_txack(TransactionType *tx)
 			return;
 
 		case STAGE_REQUEST_SEGWIT_INPUT:
+			progress = 500 + ((signatures * progress_step) >> PROGRESS_PRECISION);
```

Comment:
I should remove this line. A similar line appears below in the right path.

We  don't want to update progress for segwit, where there is no progress for inputs (progress is counted in signatures and signatures are part  of the witnesses that come later).